### PR TITLE
Updated compute modules to use `existing` und `name` input parameter

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Updated compute modules to use `existing` und `name` input parameter
- Updated parameter files
- Updated docs

Pipeline reference
[Compute: Availabilitysets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.availabilitysets.yml)
[Compute: Diskencryptionsets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.diskencryptionsets.yml)
[Compute: Galleries](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.galleries.yml)
[Compute: Images](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.images.yml)
[Compute: ProximityPlacementGroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.proximityplacementgroups.yml)
[Compute: Virtualmachines](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachines.yml)
[Compute: Virtualmachinescalesets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachinescalesets.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.Compute/virtualMachines/readme.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ This module deploys one Virtual Machine with one or multiple nics and optionally
 | `windowsConfiguration` | object | `{object}` |  | Optional. Specifies Windows operating system settings on the virtual machine. |
 | `windowsScriptExtensionCommandToExecute` | secureString |  |  | Optional. Specifies the command that should be run on a Windows VM. |
 | `windowsScriptExtensionFileData` | array | `[]` |  | Optional. Array of objects that specifies URIs and the storageAccountId of the scripts that need to be downloaded and run by the Custom Script Extension on a Windows VM. |
-| `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
+| `workspaceId` | string |  |  | Optional. Resource identifier of log analytics. |
```
Comment (eriqua) [inline arm/Microsoft.Compute/virtualMachines/readme.md:93]: Why not ID?

Comment (MrMCake) [reply arm/Microsoft.Compute/virtualMachines/readme.md]: Updated

File: arm/Microsoft.Compute/virtualMachineScaleSets/readme.md [hunk 2]
```diff
@@ -104,16 +105,15 @@ The following resources are required to be able to deploy this resource.
...
 | `vmNamePrefix` | string | `vmssvm` |  | Optional. Specifies the computer name prefix for all of the virtual machines in the scale set. |
 | `vmPriority` | string | `Regular` | `[Regular, Low, Spot]` | Optional. Specifies the priority for the virtual machine. |
-| `vmssName` | string |  |  | Optional. Name of the VMSS. |
 | `windowsScriptExtensionCommandToExecute` | secureString |  |  | Optional. Specifies the command that should be run on a Windows VM. |
 | `windowsScriptExtensionFileData` | array | `[]` |  | Optional. Array of objects that specifies URIs and the storageAccountId of the scripts that need to be downloaded and run by the Custom Script Extension on a Windows VM. |
 | `winRMListeners` | object | `{object}` |  | Optional. Specifies the Windows Remote Management listeners. This enables remote Windows PowerShell. - WinRMConfiguration object. |
-| `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
+| `workspaceId` | string |  |  | Optional. Resource identifier of log analytics. |
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.Compute/virtualMachineScaleSets/readme.md]: Updated

File: arm/Microsoft.ContainerRegistry/registries/readme.md [hunk 3]
```diff
@@ -41,6 +41,71 @@ Azure Container Registry is a managed, private Docker registry service based on
 | `trustPolicyStatus` | string |  |  | Optional. The value that indicates whether the policy is enabled or not. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `privateEndpoints`
```
Comment (MariusStorhaug) [inline arm/Microsoft.ContainerRegistry/registries/readme.md]: Double up. Remove one of the privateEndpoints

File: arm/Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/readme.md [hunk 4]
```diff
@@ -19,6 +19,23 @@
...
+        "CostCenter": "7890",
+        "ServiceName": "DeploymentValidation",
+        "Role": "DeploymentValidation"
+    }
+}
+```
+
 ### Parameter Usage: `<ParameterPlaceholder>`
 
 // TODO: Fill in Parameter usage
```
Comment (MariusStorhaug) [inline arm/Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/readme.md:41]: Remove

File: arm/Microsoft.EventGrid/topics/readme.md [hunk 5]
```diff
@@ -38,6 +38,42 @@ This module deploys Event Grid
 
 To use Private Endpoint the following dependencies must be deployed:
 
+- Destination subnet must be created with the following configuration option - `"privateEndpointNetworkPolicies": "Disabled"`.  Setting this option acknowledges that NSG rules are not applied to Private Endpoints (this capability is coming soon). A full example is available in the Virtual Network Module.
```
Comment (MariusStorhaug) [inline arm/Microsoft.EventGrid/topics/readme.md]: Double up. Remove one of the privateEndpoints

File: arm/Microsoft.Insights/privateLinkScopes/readme.md [hunk 6]
```diff
@@ -26,6 +26,42 @@ This module deploys Azure Monitor Private Link Scope
 | `scopedResources` | array | `[]` |  | Optional. Configuration Details for Azure Monitor Resources. |
 | `tags` | object | `{object}` |  | Optional. Resource tags. |
 
+### Parameter Usage: `privateEndpoints`
```
Comment (MariusStorhaug) [inline arm/Microsoft.Insights/privateLinkScopes/readme.md]: Double up. Remove one of the privateEndpoints

File: arm/Microsoft.MachineLearningServices/workspaces/readme.md [hunk 7]
```diff
@@ -39,6 +39,42 @@ This module deploys a Machine Learning Services Workspace.
 | `tags` | object | `{object}` |  | Optional. Resource tags. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `privateEndpoints`
```
Comment (MariusStorhaug) [inline arm/Microsoft.MachineLearningServices/workspaces/readme.md]: Double up. Remove one of the privateEndpoints

File: arm/Microsoft.Network/privateDnsZones/virtualNetworkLinks/readme.md [hunk 8]
```diff
@@ -14,11 +14,11 @@ This module deploys private dns zone virtual network links.
 | :-- | :-- | :-- | :-- | :-- |
 | `cuaId` | string |  |  | Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered |
 | `location` | string | `global` |  | Optional. The location of the PrivateDNSZone. Should be global. |
-| `name` | string | `[last(split(parameters('virtualNetworkId'), '/'))]` |  | Optional. The name of the virtual network link. |
+| `name` | string | `[last(split(parameters('virtualNetworkResourceId'), '/'))]` |  | Optional. The name of the virtual network link. |
 | `privateDnsZoneName` | string |  |  | Required. Private DNS zone name. |
 | `registrationEnabled` | bool |  |  | Optional. Is auto-registration of virtual machine records in the virtual network in the Private DNS zone enabled? |
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
-| `virtualNetworkResourceId` | string |  |  | Required. Link to another virtual network resource ID. |
+| `virtualNetworkResourceId` | string |  |  | Required. Link to another virtual network ID. |
```
Comment (MariusStorhaug) [inline arm/Microsoft.Network/privateDnsZones/virtualNetworkLinks/readme.md:21]: This needs a fix -> should be updated in the bicep file to be "resource ID".

File: arm/Microsoft.Network/virtualNetworks/readme.md [hunk 9]
```diff
@@ -154,8 +154,8 @@ The network security group and route table resources must reside in the same res
 
 | Output Name | Type | Description |
 | :-- | :-- | :-- |
-| `subnetResourceIds` | array | The resource IDs of the deployed subnets |
-| `subnetNames` | array | The names of the deployed subnets |
+| `subnetNames` | array | The resource IDs of the deployed subnets |
+| `subnetResourceIds` | array | The names of the deployed subnets |
```
Comment (MariusStorhaug) [inline arm/Microsoft.Network/virtualNetworks/readme.md:158]: Something is wrong here. subnetNames = resource IDs, and subnetResourceIDs = names? :D

File: arm/Microsoft.ServiceBus/namespaces/readme.md [hunk 10]
```diff
@@ -48,6 +48,42 @@ This module deploys Service Bus Namespace resource.
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 | `zoneRedundant` | bool |  |  | Optional. Enabling this property creates a Premium Service Bus Namespace in regions supported availability zones. |
 
+### Parameter Usage: `privateEndpoints`
```
Comment (MariusStorhaug) [inline arm/Microsoft.ServiceBus/namespaces/readme.md]: Double up. Remove one of the privateEndpoints

File: arm/Microsoft.Storage/storageAccounts/queueServices/queues/readme.md [hunk 11]
```diff
@@ -20,6 +20,35 @@ This module deployes a storage account queue
 | `roleAssignments` | array | `[]` |  | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or it's fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11' |
 | `storageAccountName` | string |  |  | Required. Name of the Storage Account. |
 
+### Parameter Usage: `roleAssignments`
```
Comment (MariusStorhaug) [inline arm/Microsoft.Storage/storageAccounts/queueServices/queues/readme.md:23]: Needs cleanup, seems this was missing the heading before.

File: arm/Microsoft.Storage/storageAccounts/readme.md [hunk 12]
```diff
@@ -53,6 +53,42 @@ The default parameter values are based on the needs of deploying a diagnostic st
 | `userAssignedIdentities` | object | `{object}` |  | Optional. Mandatory 'managedServiceIdentity' contains UserAssigned. The identy to assign to the resource. |
 | `vNetId` | string |  |  | Optional. Virtual Network Identifier used to create a service endpoint. |
 
+### Parameter Usage: `privateEndpoints`
```
Comment (MariusStorhaug) [inline arm/Microsoft.Storage/storageAccounts/readme.md]: Double up. Remove one of the privateEndpoints

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]

File: arm/Microsoft.ContainerRegistry/registries/readme.md (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.ContainerRegistry/registries/readme.md]: done

File: arm/Microsoft.EventGrid/topics/readme.md (hunk 5 above)
Comment (MrMCake) [reply arm/Microsoft.EventGrid/topics/readme.md]: done

File: arm/Microsoft.CognitiveServices/accounts/readme.md [hunk 13]
```diff
@@ -38,6 +38,42 @@ This module deploys different kinds of Cognitive Services resources
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `privateEndpoints`
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.CognitiveServices/accounts/readme.md]: done

File: arm/Microsoft.Automation/automationAccounts/readme.md [hunk 14]
```diff
@@ -51,6 +51,42 @@ This module deploys an Azure Automation Account.
 | `variables` | _[variables](variables/readme.md)_ array | `[]` |  | Optional. List of variables to be created in the automation account. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `privateEndpoints`
```
[the remark this answers is not in the document]
Comment (MrMCake) [reply arm/Microsoft.Automation/automationAccounts/readme.md]: done

File: arm/Microsoft.Insights/privateLinkScopes/readme.md (hunk 6 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/privateLinkScopes/readme.md]: done

File: arm/Microsoft.MachineLearningServices/workspaces/readme.md (hunk 7 above)
Comment (MrMCake) [reply arm/Microsoft.MachineLearningServices/workspaces/readme.md]: done

File: arm/Microsoft.ServiceBus/namespaces/readme.md (hunk 10 above)
Comment (MrMCake) [reply arm/Microsoft.ServiceBus/namespaces/readme.md]: done

File: arm/Microsoft.Storage/storageAccounts/readme.md (hunk 12 above)
Comment (MrMCake) [reply arm/Microsoft.Storage/storageAccounts/readme.md]: done

File: arm/Microsoft.Network/privateDnsZones/virtualNetworkLinks/readme.md (hunk 8 above)
Comment (MrMCake) [reply arm/Microsoft.Network/privateDnsZones/virtualNetworkLinks/readme.md]: done

File: arm/Microsoft.Network/virtualNetworks/readme.md (hunk 9 above)
Comment (MrMCake) [reply arm/Microsoft.Network/virtualNetworks/readme.md]: Fixed

File: arm/Microsoft.Storage/storageAccounts/queueServices/queues/readme.md (hunk 11 above)
Comment (MrMCake) [reply arm/Microsoft.Storage/storageAccounts/queueServices/queues/readme.md:23]: done

File: arm/Microsoft.ContainerRegistry/registries/readme.md [hunk 15]
```diff
@@ -41,6 +41,35 @@ Azure Container Registry is a managed, private Docker registry service based on
 | `trustPolicyStatus` | string |  |  | Optional. The value that indicates whether the policy is enabled or not. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `roleAssignments`
+
+```json
+"roleAssignments": {
+    "value": [
+        {
+            "roleDefinitionIdOrName": "Desktop Virtualization User",
```
Comment (eriqua) [inline arm/Microsoft.ContainerRegistry/registries/readme.md:50]: Remove first element with Desktop Virtualization User

File: arm/Microsoft.ContainerService/managedClusters/readme.md [hunk 16]
```diff
@@ -71,6 +71,35 @@ This module deploys Azure Kubernetes Cluster (AKS).
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `roleAssignments`
+
+```json
+"roleAssignments": {
+    "value": [
+        {
+            "roleDefinitionIdOrName": "Desktop Virtualization User",
```
Comment (eriqua) [inline arm/Microsoft.ContainerService/managedClusters/readme.md:80]: Remove first element with Desktop Virtualization User

File: arm/Microsoft.DocumentDB/databaseAccounts/readme.md [hunk 17]
```diff
@@ -44,6 +44,52 @@ This module deploys a Documentdb database account and its child resources.
 | `userAssignedIdentities` | object | `{object}` |  | Optional. Mandatory if 'managedServiceIdentity' contains UserAssigned. The list of user identities associated with the database account. |
 | `workspaceId` | string |  |  | Optional. Resource ID of log analytics. |
 
+### Parameter Usage: `roleAssignments`
+
+```json
+"roleAssignments": {
+    "value": [
+        {
+            "roleDefinitionIdOrName": "Desktop Virtualization User",
```
Comment (eriqua) [inline arm/Microsoft.DocumentDB/databaseAccounts/readme.md:53]: Remove first element with Desktop Virtualization User

File: arm/Microsoft.EventHub/namespaces/readme.md [hunk 18]
```diff
@@ -46,6 +46,71 @@ This module deploys an EventHub namespace
...
+}
+```
+
+### Parameter Usage: `roleAssignments`
+
+```json
+"roleAssignments": {
+    "value": [
+        {
+            "roleDefinitionIdOrName": "Desktop Virtualization User",
```
Comment (eriqua) [inline arm/Microsoft.EventHub/namespaces/readme.md:91]: Remove first element with Desktop Virtualization User

File: arm/Microsoft.Web/connections/readme.md [hunk 19]
```diff
@@ -31,6 +31,35 @@ This module deploys an Azure API Connection.
 | `tags` | object | `{object}` |  | Optional. Tags of the resource. |
 | `testLinks` | array | `[]` |  | Optional. Links to test the API connection. |
 
+### Parameter Usage: `roleAssignments`
+
+```json
+"roleAssignments": {
+    "value": [
+        {
+            "roleDefinitionIdOrName": "Desktop Virtualization User",
```
Comment (eriqua) [inline arm/Microsoft.Web/connections/readme.md:40]: Remove first element with Desktop Virtualization User

File: arm/Microsoft.Web/sites/readme.md [hunk 20]
```diff
@@ -49,6 +49,42 @@ This module deploys a Web or Function App
...
+
+- Destination subnet must be created with the following configuration option - `"privateEndpointNetworkPolicies": "Disabled"`.  Setting this option acknowledges that NSG rules are not applied to Private Endpoints (this capability is coming soon). A full example is available in the Virtual Network Module.
+- Although not strictly required, it is highly recommended to first create a private DNS Zone to host Private Endpoint DNS records. See [Azure Private Endpoint DNS configuration](https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns) for more information.
+
+```json
+"privateEndpoints": {
+    "value": [
+        // Example showing all available fields
+        {
+            "name": "sxx-az-sa-cac-y-123-pe", // Optional: Name will be automatically generated if one is not provided here
```
Comment (eriqua) [inline arm/Microsoft.Web/sites/readme.md:64]: weird name, let's maybe find a generic one to copy over

File: arm/Microsoft.ContainerRegistry/registries/readme.md (hunk 15 above)
Comment (MrMCake) [reply arm/Microsoft.ContainerRegistry/registries/readme.md]: done

File: arm/Microsoft.DocumentDB/databaseAccounts/readme.md (hunk 17 above)
Comment (MrMCake) [reply arm/Microsoft.DocumentDB/databaseAccounts/readme.md]: done

File: arm/Microsoft.ContainerService/managedClusters/readme.md (hunk 16 above)
Comment (MrMCake) [reply arm/Microsoft.ContainerService/managedClusters/readme.md]: done

File: arm/Microsoft.EventHub/namespaces/readme.md (hunk 18 above)
Comment (MrMCake) [reply arm/Microsoft.EventHub/namespaces/readme.md]: done

File: arm/Microsoft.Web/connections/readme.md (hunk 19 above)
Comment (MrMCake) [reply arm/Microsoft.Web/connections/readme.md]: done

**Review by eriqua — APPROVED** [review]
🆗

File: arm/Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/readme.md (hunk 4 above)
Comment (MrMCake) [reply arm/Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers/readme.md]: done

**Review by MariusStorhaug — APPROVED** [review]
Approved

**Pull request merged**